### ISSUE TITLE
aws_codedeploy_deployment_group green_fleet_provisioning_option COPY_AUTO_SCALING_GROUP got error

### ISSUE THREAD

**Issue by relifeted** [terraform-providers/terraform-provider-aws#2233]
### Terraform Version

0.10.8

### Affected Resource(s)

- aws_codedeploy_deployment_group

### Terraform Configuration Files
```hcl

resource "aws_codedeploy_deployment_group" "CodeDeploy" {
  app_name               = "${aws_codedeploy_app.app.name}"
  deployment_group_name  = "${var.group_name}"
  service_role_arn       = "${aws_iam_role.CodeDeploy.arn}"
  deployment_config_name = "CodeDeployDefault.AllAtOnce"

  deployment_style {
    deployment_option = "WITH_TRAFFIC_CONTROL"
    deployment_type   = "BLUE_GREEN"
  }

  load_balancer_info {
    target_group_info {
      name = "${var.target_group_name}"
    }
  }

  blue_green_deployment_config {
    deployment_ready_option {
      action_on_timeout = "CONTINUE_DEPLOYMENT"
    }

    green_fleet_provisioning_option {
      action = "COPY_AUTO_SCALING_GROUP"
    }

    terminate_blue_instances_on_deployment_success {
      action = "TERMINATE"
    }
  }

  autoscaling_groups = ["${var.autoscaling_groups}"]

}
```

### Expected Behavior

create or update CodeDeploy Deployment Group with "Copy Auto Scaling Group" provision option

### Actual Behavior

got error: InvalidBlueGreenDeploymentConfigurationException: Exactly one AutoScaling group must be specifed when selecting the COPY_AUTO_SCALING_GROUP green fleet provisioning option.

### Steps to Reproduce

1. Create a deployment group as above
2. run terraform plan
3. run terraform apply

**Comment by joao-carreira** [comment]
Hi!
Any updates regarding this topic?

**Comment by hbarcelos** [comment]
+1

**Comment by en0x** [comment]
+1

**Comment by Telling** [comment]
I too just ran into this problem :/

**Comment by AlexBHarley** [comment]
+1

**Comment by officel** [comment]
+1
```
Terraform v0.11.3
+ provider.aws v1.8.0

ex) terraform apply
aws_codedeploy_deployment_group.this: Modifying... (ID: 0e8c1269-b37d-418e-9ac1-fcf312fXXXXX)
  blue_green_deployment_config.0.terminate_blue_instances_on_deployment_success.0.termination_wait_time_in_minutes: "10" => "5"

Error: Error applying plan:

1 error(s) occurred:

* aws_codedeploy_deployment_group.this: 1 error(s) occurred:

* aws_codedeploy_deployment_group.this: InvalidBlueGreenDeploymentConfigurationException: Exactly one AutoScaling group must be specifed when selecting the COPY_AUTO_SCALING_GROUP green fleet provisioning option.
        status code: 400, request id: 4fb082b1-0ce2-11e8-87bb-fba4b39XXXXX

Oh..._no
```

**Comment by ronster37** [comment]
+1
```
Terraform v0.11.3
provider.aws 1.9
```
Anyone have any workarounds?

**Comment by JimtotheB** [comment]
There are all kinds of problems with using Terraform for Codedeploy blue/green. A major one which I think is faced here. When you do a deployment Codeploy creates a new ASG and deleted the one that Terraform knows about. 

I have tried doing a `terraform refresh` followed by importing the newly created ASG into the state, but I still get the error referenced above.

You have to import the ASG in any case to destroy the stack.

**Comment by thatrevguy** [comment]
Just ran into this as well 
```
Terraform v0.11.3
provider.aws 1.9
```

**Comment by sharath-yml** [comment]
+1
I had this issue, and a crude workaround for this was manually setting the "Environment Configuration" to "Automatically copy Auto Scaling group" through AWS console, and doing a "terraform refresh".

Please review this pull request soon.

**Comment by Just-en-Time** [comment]
+1, still encountering this on
`
Terraform v0.11.5
+ provider.aws v1.13.0
`

**Comment by Just-en-Time** [comment]
+1, still encountering this on Terraform v0.11.5, provider.aws v1.13.0

**Comment by Just-en-Time** [comment]
+1, hoping to keep deployment process automated

**Comment by Jonahss** [comment]
I'm guessing that this issue may be solved if #4678

**Comment by luisrod-olx** [comment]
Hi all. This is an issue since last year, any news on this?

**Comment by niclic** [comment]
I am working on a fix for the _original_ issue reported here. Should have it pushed up in the next day or so. I apologize for any inconvenience. The issue reported in [#4678](https://github.com/terraform-providers/terraform-provider-aws/issues/4678) is a different matter altogether. I will address that separately.

**Comment by macnibblet** [comment]
@niclic How's that push coming along? I'm also experiencing this problem

**Comment by niclic** [comment]
@macnibblet I submitted the fix last week. It is [#5827](https://github.com/terraform-providers/terraform-provider-aws/pull/5827).

I can't say when it will be reviewed & merged (these things can sometimes take a frustratingly long time) but up-votes do help. Thank you.

**Comment by bflad** [comment]
The fix for the original issue has been merged and will release with version 1.39.0 of the AWS provider, likely middle of next week.
